samples/nrf9160/gps/src/main.c

```diff
@@ -98,40 +98,49 @@ static int setup_modem(void)
 }
 
 #ifdef CONFIG_SUPL_CLIENT_LIB
-/* Accepted network statuses read from modem */
-static const char status_reg_home[]    = "+CEREG: 1";
-static const char status_reg_roaming[] = "+CEREG: 5";
-
-static void wait_for_lte(void *context, const char *response)
+BUILD_ASSERT(IS_ENABLED(CONFIG_LTE_NETWORK_MODE_LTE_M_GPS) ||
+	     IS_ENABLED(CONFIG_LTE_NETWORK_MODE_NBIOT_GPS) ||
+	     IS_ENABLED(CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT_GPS),
+	     "To use SUPL and GPS, CONFIG_LTE_NETWORK_MODE_LTE_M_GPS, "
+	     "CONFIG_LTE_NETWORK_MODE_NBIOT_GPS or "
+	     "CONFIG_LTE_NETWORK_MODE_LTE_M_NBIOT_GPS must be enabled");
+
+static void lte_handler(const struct lte_lc_evt *const evt)
 {
-	if (!memcmp(status_reg_home, response, AT_CMD_SIZE(status_reg_home)) ||
-	    !memcmp(status_reg_roaming, response, AT_CMD_SIZE(status_reg_roaming))) {
-		k_sem_give(&lte_ready);
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS:
+		if ((evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME) ||
+		    (evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_ROAMING)) {
+			printk("Connected to LTE network\n");
```

Comment:
There's currently two traces when LTE is connected, this one and a second one in `get_agps_data()` function. One of those should be removed, I suggest you remove the "Established LTE link" trace in `get_agps_data()`.

Reply:
Removed the one in get_agps_data